dev/tasks/tasks.yml

```diff
@@ -1789,6 +1789,18 @@ tasks:
       r_image: r-base
       r_tag: 3.6-opensuse42
       not_cran: "TRUE"
+  
+  test-r-without-dataset-parquet-s3:
```

Comment:
I actually liked the "minimal" name, matches the cpp minimal build example we test. If you're worried about confusion with the `LIBARROW_MINIMAL` env var (I'm not personally), we could try to make that var be truly minimal. I think currently we only ever check for whether it is set to `false`, to mean "build all the compression libs etc.", but we don't check if it is `true`. So we could make `true` be truly minimal, without these optional features, and the default (unset) means the current default of all features on but not compression and S3.

On an unrelated note, you should add this crossbow job to the `r` and `nightly` groups so that it gets run regularly.

Reply:
> S3 is on because linuxlibs.R is turning it on: https://dev.azure.com/ursacomputing/crossbow/_build/results?buildId=1741&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=6c939d89-0d1a-51f2-8b30-091a7a82e98c&l=24
> 
> I think you'll find the issue pretty quickly but LMK if you need a pointer.

Looks like `configure` is setting `LIBARROW_MINIMAL=false` because `NOT_CRAN` is `true`:
```sh
if [ "${LIBARROW_MINIMAL}" = "" ] && [ "${NOT_CRAN}" = "true" ]; then
  LIBARROW_MINIMAL=false; export LIBARROW_MINIMAL
fi
```
and then `linuxlibs.r` sets `arrow_s3` to `TRUE`:
```r
arrow_s3 <- toupper(Sys.getenv("ARROW_S3")) == "ON" || tolower(Sys.getenv("LIBARROW_MINIMAL")) == "false"
```
I will try to solve this by explicitly setting `LIBARROW_MINIMAL=true` in this build.

Reply:
> I actually liked the "minimal" name, matches the cpp minimal build example we test. If you're worried about confusion with the `LIBARROW_MINIMAL` env var (I'm not personally), we could try to make that var be truly minimal. I think currently we only ever check for whether it is set to `false`, to mean "build all the compression libs etc.", but we don't check if it is `true`. So we could make `true` be truly minimal, without these optional features, and the default (unset) means the current default of all features on but not compression and S3.
> 
> On an unrelated note, you should add this crossbow job to the `r` and `nightly` groups so that it gets run regularly.

For now I think I'll just set `LIBARROW_MINIMAL=true` so the "minimal" name is not misleading. Later we might consider whether an explicit `LIBARROW_MINIMAL=true` should toggle Dataset and Parquet off.

Reply:
The `r` and `nightly` groups are defined with wildcards that will match the name I used for this test.